api/users/models.py

```diff
@@ -121,6 +121,18 @@ def __str__(self):
     def subscribe_to_mailing_list(self):
         mailer_lite.subscribe(self)
 
+    @hook(BEFORE_DELETE)
+    def delete_organisation_with_a_last_user(self):
+        user_organisations = UserOrganisation.objects.filter(
+            user_id=self.initial_value("id")
+        )
+        for user_organisation in user_organisations:
+            users_count = UserOrganisation.objects.filter(
```

Comment:
This can be optimized, e.g. something like:

```python
organisation_ids_to_delete = self.organisations.filter(user=self)\
    .annotate(num_users=Count("users"))\
    .filter(num_users=1)\
    .values_list("id", flat=True)
```

Reply:
The proposed solution can't be implemented as is since organizations is a many_to_many field. However the code has been simplified in a similar way.